R/parse-query.R

```diff
@@ -179,3 +151,78 @@ extractPathParams <- function(def, path){
 
   vals
 }
+
+
+#' combine args that share the same name
+#' @noRd
+combine_keys <- function(obj, type) {
+
+  keys <- names(obj)
+  unique_keys <- unique(keys)
+
+  if (length(unique_keys) == length(keys) || is.null(keys)) {
+    return(obj)
+  }
+
+  vals <- unname(obj)
+
+  extra_args <- NULL
+  if (type == "multi") {
+    # handle unnamed args by removing them from being merged and adding them back again at the end
+    no_name_positions <- (keys == "")
+    if (any(no_name_positions)) {
+      extra_args <- vals[no_name_positions]
+      vals <- vals[!no_name_positions]
+      keys <- keys[!no_name_positions]
+      unique_keys <- setdiff(unique_keys, "")
+    }
+  }
+
+  cleanup_item <- switch(
+    type,
+    "query" =
+      function(x) {
+        unname(unlist(x))
+      },
+    "multi" =
+      function(x) {
+        if (length(x) == 1) {
+          # return first item only
+          x[[1]]
+        } else {
+          # return list of internal named items
+          # aka... unlist the top layer only. Maintain the inner layer names
+          as.list(unlist(x, recursive = FALSE))
```

Comment:
Add in a test for this logic (maybe `x` of `list(A = list(x = 1, y = 3), A = 2, A = 5))`)

```suggestion
          as.list(unlist(unname(x), recursive = FALSE))
```